### ISSUE TITLE
LLVM 17 regression: floating point ops on `f80` and `f128` behavior tests

### ISSUE THREAD

**Issue by andrewrk** [ziglang/zig#16844]
### Zig Version

0.12.0-dev.116+4cc2544cd

### Steps to Reproduce and Observed Behavior

```
$ zig build test-behavior -fqemu
run test behavior-powerpc64le-linux-none-Debug: error: while executing test 'test.@floor f80', the following command terminated with signal 11 (expected exited with code 0):
qemu-ppc64le /home/andy/misc/zig/zig-cache/o/9ea17d1e70534cb3892eb0bef0326bd6/test --listen=- 
run test behavior-aarch64-linux-none-Debug: error: while executing test 'test.@floor f80', the following command terminated with signal 15 (expected exited with code 0):
qemu-aarch64 /home/andy/misc/zig/zig-cache/o/683f4ea11364b984206e2a5859e33143/test --listen=- 
run test behavior-aarch64-linux-musl-Debug-libc: error: while executing test 'test.@floor f80', the following command terminated with signal 9 (expected exited with code 0):
```

For the aarch64 tests, they entered an infinite loop so I had to kill them.

Same deal with `@floor f128` behavior test:

```
run test behavior-powerpc64le-linux-none-Debug: error: while executing test 'test.@floor f128', the following command terminated with signal 11 (expected exited with code 0):
qemu-ppc64le /home/andy/misc/zig/zig-cache/o/ad107966f2fda47955afc0989e32ca12/test --listen=- 
run test behavior-aarch64-linux-musl-Debug-libc: error: while executing test 'test.@floor f128', the following command terminated with signal 15 (expected exited with code 0):
qemu-aarch64 /home/andy/misc/zig/zig-cache/o/5e15200f207601f61d45026e1b97d3e1/test --listen=- 
run test behavior-aarch64-linux-none-Debug: error: while executing test 'test.@floor f128', the following command terminated with signal 15 (expected exited with code 0):
qemu-aarch64 /home/andy/misc/zig/zig-cache/o/308911d720954751c132af6a2216f44a/test --listen=- 
```

Same deal with aarch64 going into an infinite loop.

### Expected Behavior

All tests passed.

-----

Related: #12602 

Next step is to file an upstream bug report with an LLVM IR reduction. Hopefully it can get fixed before the final 17 tag.

**Comment by andrewrk** [comment]
Clue: seems to be something related to "none" vs "musl" or "gnu" ABIs.

**Comment by jacobly0** [comment]
Fixed by 6509436c23433d80fd3b3803f5c68b235958e7a8
